### ISSUE TITLE
[Debug] Execution stops in a TypeScript source file at a code line without a breakpoint and debugger statement

### ISSUE THREAD

**Issue by Deilan** [Microsoft/vscode#19676]
- VSCode Version: 1.8.1
- OS Version: Windows 8.1 x64

Execution stops in a TypeScript source file at a code line without a breakpoint and debugger statement. Actually there was once a breakpoint before, but I removed it. Nevertheless the execution continues to stop at that line. Tried to reset breakpoint there, rebuild the code, restart VS Code - doesn't help.

![untitled](https://cloud.githubusercontent.com/assets/4777512/22513304/e33dad22-e8ac-11e6-9259-c9f868cf4c80.png)

Don't know how to reproduce this with an arbitrary code line, but it stable reproduces with this line.

If you need some additional info - let me know.

Contents of `.vscode\launch.json`:
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "main",
            "cwd": "${workspaceRoot}",
            "program": "${workspaceRoot}/src/app.ts",
            "outFiles": [
                "${workspaceRoot}/built/**/*.js"
            ],
            "sourceMaps": true,
            "smartStep": true,
            "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/electron.cmd",
            "runtimeArgs": [
                "${workspaceRoot}",
                "--enable-logging",
                "--remote-debugging-port=9222"
            ],
            "preLaunchTask": "build"
        },
        {
            "type": "chrome",
            "request": "attach",
            "name": "renderer",
            "port": 9222,
            "webRoot": "${workspaceRoot}/built",
            "sourceMaps": true,
            "smartStep": true
        }
    ],
    "compounds": [
        {
            "name": "multi",
            "configurations": [
                "main",
                "renderer"
            ]
        }
    ]
}
```

**Comment by isidorn** [comment]
@Deilan if you open the debug pane - breakpoints view do you see some breakpoints there? If you trigger a 'remove all breakpoints' actions does the program still stop?
Does the program still stop if you change `"type"` in your `launch.json` to `"node2"`

**Comment by Deilan** [comment]
@isidorn

The program does not start with `"type"` set to `"node2"` in `launch.json` because of the following error:

> Cannot connect to runtime process, timeout after 10000 ms - (reason: Cannot connect to the target: connect ECONNREFUSED 127.0.0.1:8816).

Wonder why. As you already may mention I try to debug an electron application.

**Comment by Deilan** [comment]
@isidorn 

I've figured out that the problem seems to be with file naming. Let's me show you the images:

The breakpoint is actually set before the debugging session:

![untitled](https://cloud.githubusercontent.com/assets/4777512/22515783/bf46af2e-e8b4-11e6-98e2-9e20b18b1aa7.png)


After that I close the file (the breakpoint is still there) and fire off the debug session. The breakpoint gets hit, but the file name's casing in the editor is wrong:

![untitled2](https://cloud.githubusercontent.com/assets/4777512/22515767/af53c7fa-e8b4-11e6-8dd7-e6ea0c0f853d.png)

Now if I set the breakpoint manually there, I have two different program.ts files with different casings in breakpoint view:

![untitled3](https://cloud.githubusercontent.com/assets/4777512/22515877/147ccc44-e8b5-11e6-915a-1f392d823178.png)

If I unset the breakpoint, the one with the file capital letter is gone, the original is still there.
